File: erpnext/controllers/item_price.py
import frappe

@frappe.whitelist()
def get_price_list_rate_based_on_rate_in_strip(rate_in_strip, item_code):
    item_doc = frappe.get_doc("Item", item_code)
    if(len(item_doc.uoms)>1 and rate_in_strip):
        lastElement = item_doc.uoms[len(item_doc.uoms)-1]
        conversion_factor = lastElement.conversion_factor
        if not conversion_factor:
            conversion_factor=1
            # print(item_doc.name+" Updated Succesfully with coversion_factor = 1")
        return { 'price_list_rate': float(rate_in_strip)/float(conversion_factor) }
    else:
        return {  'price_list_rate': float(rate_in_strip) }
    

@frappe.whitelist()
def item_price_on_validate(doc, method):
    # print("on validate")
    item_doc = frappe.get_doc("Item", doc.item_code)
    if(len(item_doc.uoms)>1 and doc.rate_in_strip):
        lastElement = item_doc.uoms[len(item_doc.uoms)-1]
        conversion_factor = lastElement.conversion_factor
        if not conversion_factor:
            conversion_factor=1
            # print(item_doc.name+" Updated Succesfully with coversion_factor = 1")
        doc.price_list_rate = float(doc.rate_in_strip)/float(conversion_factor)
    else:
        doc.price_list_rate = float(doc.rate_in_strip)
    frappe.db.commit()

@frappe.whitelist()
def get_item_conversion_factor(item_code, mrp_in_strip):
    item_doc = frappe.get_doc("Item", item_code)
    if(len(item_doc.uoms)>1 and mrp_in_strip):
        lastElement = item_doc.uoms[len(item_doc.uoms)-1]
        conversion_factor = lastElement.conversion_factor
        if not conversion_factor:
            conversion_factor=1
            # print(item_doc.name+" Updated Succesfully with coversion_factor = 1")
        return { 'mrp': float(mrp_in_strip)/float(conversion_factor) }
    else:
        return {  'mrp': float(mrp_in_strip) }

File: erpnext/public/js/webeaz/item_price.js
frappe.ui.form.on('Item Price',{
    rate_in_strip(frm){
      get_price_list_rate_based_on_rate_in_strip(frm)
    },
    validate: function(frm){
      get_price_list_rate_based_on_rate_in_strip(frm)
    },
    refresh: function(frm){
      frm.set_df_property('price_list_rate', 'read_only', 1)
    }
});
  
var get_price_list_rate_based_on_rate_in_strip = function(frm){
  if(frm.doc.rate_in_strip && frm.doc.item_code){
    frappe.call({
      "method": "erpnext.controllers.item_price.get_price_list_rate_based_on_rate_in_strip",
      "args": {
        'rate_in_strip': frm.doc.rate_in_strip,
        'item_code': frm.doc.item_code
      },
      callback: function(r){
        if(r.message){
          frm.set_value('price_list_rate', r.message.price_list_rate);
          // frm.set_df_property('price_list_rate', 'read_only', 1)

        }
      }
    });
  }
}